File: Java-CodeServer/src/main/resources/js/prompto/type/NullType.js
var BaseType = require("./BaseType").BaseType;

function NullType() {
    BaseType.call(this, "Null");
    return this;
}

NullType.prototype = Object.create(BaseType.prototype);
NullType.prototype.constructor = NullType;

NullType.instance = new NullType();

NullType.prototype.checkUnique = function(self, context) {
    // ok
};

NullType.prototype.checkExists = function(self, context) {
    // ok
};

NullType.prototype.isAssignableTo = function(self, context, other) {
    return true;
};

NullType.prototype.isMoreSpecificThan = function(self, context, other) {
    return false;
};

exports.NullType = NullType;

File: Java-Core/src/main/java/prompto/type/NullType.java
package prompto.type;

import java.lang.reflect.Type;

import prompto.error.SyntaxError;
import prompto.runtime.Context;

public class NullType extends BaseType {

	static NullType instance = new NullType();
	
	public static NullType instance() {
		return instance;
	}
	
	private NullType() {
		super(Family.NULL);
	}

	@Override
	public Type getJavaType() {
		return null;
	}

	@Override
	public void checkUnique(Context context) throws SyntaxError {
		// ok
	}

	@Override
	public void checkExists(Context context) throws SyntaxError {
		// ok
	}

	@Override
	public boolean isAssignableTo(Context context, IType other) {
		return true;
	}

	@Override
	public boolean isMoreSpecificThan(Context context, IType other) {
		return false;
	}
		

}
